### PR TITLE
Network utils: add fallback to fetch()

### PR DESCRIPTION
**Pull request by bershanskiy**

### CODE REVIEW

File: src/background/utils/network.ts [hunk 1]
```diff
@@ -9,22 +9,56 @@ interface RequestParams {
...
     return new Promise((resolve, reject) => {
-        const request = new XMLHttpRequest();
-        request.overrideMimeType('text/plain');
-        request.open('GET', params.url, true);
-        request.onload = () => {
-            if (request.status >= 200 && request.status < 300) {
-                resolve(request.responseText);
-            } else {
-                reject(new Error(`${request.status}: ${request.statusText}`));
+        if (XMLHttpRequest) {
```
Comment (Gusted) [inline src/background/utils/network.ts:12]: Can we move these 2 detections to `utils/platform.ts`

In the form of `const is IsXMLHttpRequestSupported = typeof XMLHttpRequest === 'function'` and `typeof isFetchSupported = typeof fetch === 'function'`.

Also should it by default prefer fetch over XMLHttpRequest?

Comment (Gusted) [reply src/background/utils/network.ts:12]: Oops, too late 🤷🏽‍♂️

Comment (bershanskiy) [reply src/background/utils/network.ts:12]: Oh, I rushed and missed your comments by one minute. I'll create a follow-up for this.

Comment (bershanskiy) [reply src/background/utils/network.ts:12]: Here it is: #6486

**Comment by bershanskiy** [comment]
RE to @Gusted:

> Also should it by default prefer fetch over XMLHttpRequest?

I decided to use `XMLHttpRequest` when available because browsers are weird about abort signals in `fetch`. Specifically, some versions of Safari can accept abort signal, but then just ignore when you actually abort it. If we use `fetch` as a fallback, we know that right now it will be used only in service workers in Chrome MV3 build and when other browsers inplement MV3 they will already support `fetch` properly.

Also, I resorted to using an extra boolean `timedOut` because browsers throw different errors when you abort a `fetch` request.

**Comment by Gusted** [comment]
> Specifically, some versions of Safari can accept abort signal, but then just ignore when you actually abort it

Ah, great. Always damn safari.

**Pull request merged**